quinn/src/tests.rs

```diff
@@ -445,6 +449,21 @@ async fn echo((mut send, recv): (SendStream, RecvStream)) {
     let _ = send.finish().await;
 }
 
+fn gen_seed() -> u64 {
+    let mut buf = [0u8; 8];
+    rand::thread_rng().fill_bytes(&mut buf[..]);
+    let seed = u64::from_be_bytes(buf);
+    println!("Using seed {:?}", seed);
+    seed
+}
```

Comment:
Isn't this just controlling the actual data? I don't see how the RNG would influence boundaries at all.

Reply:
It's just the data in this case, so it could matter if eg one run produce all `0`s and and shift in data couldn't be detected. That local `rng` is used for nothing else.

Would be nice to capture different issues by fixing nondeterminism. But there is so much of it in the system (like OS scheduling, tokio scheduling, time, etc) that it's hard to avoid it.

Reply:
Right, this layer isn't really the right place for truly deterministic tests, but I still don't see what we have to gain by adding additional, avoidable nondeterminism.

Reply:
Do you prefer to use a fixed seed? I am fine either way

Reply:
I do, yes, sorry if that was unclear :sweat_smile: